core/primitives/src/shard_layout.rs

```diff
@@ -620,6 +620,45 @@ impl ShardLayout {
         Ok(parent_shard_id)
     }
 
+    /// Derive new shard layout from an existing one
+    pub fn derive_shard_layout(
+        base_layout: &ShardLayout,
+        new_boundary_account: AccountId,
+    ) -> ShardLayout {
+        let mut boundary_accounts = base_layout.boundary_accounts().clone();
+        let mut shard_ids = base_layout.shard_ids().collect::<Vec<_>>();
+        let mut shards_split_map = shard_ids
+            .iter()
+            .map(|id| (*id, vec![*id]))
+            .collect::<BTreeMap<ShardId, Vec<ShardId>>>();
+
+        debug_assert!(
+            boundary_accounts.iter().position(|acc| acc == &new_boundary_account).is_none(),
+            "duplicated boundary account",
+        );
```

Comment:
debug_assert -> assert 
Misformed shard layout is a breaking condition, we should always panic here (not only in the debug mode).

Reply:
The reason I wanted to use debug_assert here was for better performance by sparing production code of the assertions, while being able to prevent certain bugs from slipping into production given the CI build (at least in this particular case it should?). But maybe it's not worth it.

Reply:
Fair but shard layout creation is not a performance sensitive operation. Quite the opposite actually, it is a very rare event. If the shard layout is invalid things will break anyway so it's best to fail fast and panic in all modes.